server/src/main/java/org/candlepin/resource/UserResource.java

```diff
@@ -141,31 +207,38 @@ public User createUser(@ApiParam(hidden = true) User user) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{username}")
-    public User updateUser(@PathParam("username")
-        @Verify(User.class) String username,
-        @ApiParam(name = "user", required = true) User user) {
+    public UserDTO updateUser(
+        @PathParam("username") @Verify(User.class) String username,
+        @ApiParam(name = "user", required = true) UserDTO dto) {
 
         // Note, to change the username, the old username needs to be provided.
-        if (userService.findByLogin(username) == null) {
-            throw new NotFoundException(i18n.tr("User {0} does not exist", username));
+        User user = this.fetchUserByUsername(username);
+
+        // Apparently we allow anything to change here...???
+        if (dto.getUsername() != null) {
+            user.setUsername(dto.getUsername());
         }
-        return userService.updateUser(user);
-    }
 
+        if (dto.getPassword() != null) {
+            user.setPassword(dto.getPassword());
+        }
+
+        if (dto.isSuperAdmin() != null) {
+            user.setSuperAdmin(dto.isSuperAdmin());
+        }
+
+        return this.modelTranslator.translate(userService.updateUser(user), UserDTO.class);
+    }
 
     @ApiOperation(notes = "Removes a User", value = "deleteUser")
     @ApiResponses({ @ApiResponse(code = 410, message = "") })
```

Comment:
If we're throwing `NotFoundException` instead of `GoneException` now, then we should change this to 404. Maybe even add in the 400 since it could throw `BadRequestException`.

Reply:
Done.